[WIP] Reading proxies from json

Sauromayne opened the pull request:
Builds a list of dictionaries to use for proxies reading user, pass, and ip:port from the json file.

Can be improved by enabling with flag rather than in the json and building dictionaries with separate ip:port for the http and https keys. So far I have only been able to get a proxy working using the same ip:port for both keys. I'm sure there are proxies where this wouldn't work though.

Sauromayne commented:
I have updated the code to accept a seperate ip:port for the http and https dictionary keys.

To do: enable proxies with flag rather than in the config (I'm not sure how to do this), add support for proxies that use ip auth rather than user/pass

timbru31 commented on config/amazon_requests_proxies.json:
```diff
@@ -0,0 +1,7 @@
+{
+      "enable": "n",
```
Why no real Boolean value?

timbru31 commented on config/amazon_requests_proxies.json:
```diff
@@ -0,0 +1,7 @@
+      "enable": "n",
+      "user": "user",
+      "pass": "pass",
+      "ip_port": [{"http":"XXX.XXX.XXX.XXX:XXXX", "https": "XXX.XXX.XXX.XXX:XXXX"},
```
Can you properly format this?

Sauromayne replied:
Yeah. Can I use black for that? This is the first time I have made a json file

timbru31 commented on stores/amazon_requests.py:
```diff
@@ -180,22 +180,26 @@ def __init__(
 
+        # build the proxies list
         self.proxies = []
+        for i in proxy_dict["ip_port"]:
```
This is not performant. Check for enabled True and move statements in if clause

Sauromayne replied:
I have updated the code to use flag now since I was able to figure that out in cli.py, I'm sorry I'm not sure exactly what you mean. The for loop builds the proxy dictionaries so not really sure what you mean by non performant.

Sorry I have only been working with Python for 2 months and has been a few years since I have coded before that

Sauromayne commented:
I have updated the code to support both ip and user/pass authentication methods. I also reformat the json template and added instructions. 

I don't have an ip authenticated proxy to test with but looking at the documentation I am pretty sure its correct. I think that it is ready now but @timbru31 said the for loops I am using to build the lists are non-performant so let me know.

Sauromayne commented:
added support for socks proxies
